### DISCUSSION TITLE
Use custom output directory

### DISCUSSION THREAD

**Discussion by JakobDev** [pyinstaller/pyinstaller#7375] [Help]
I build a App as directory, not with onefile. By default, PyInstaller places all files inside `dist/<scriptname>`. But I need them inside `WindowsBuild/Portable`. This needed for my CI, which is used with multiple Projects. I'm currently just copying the files, but it would be better,  if pyinstaller could directly place them in the right directory.

I had found `--distpath `, but running `pyinstaller --distpath WindowsBuild/Portable <scriptname>.py` places the files under `WindowsBuild/Portable/<scriptname>`. Now I have tried it with the `--name` argument. `pyinstaller --distpath WindowsBuild --name Portable <scriptname>.py`. This Places in the right directory, but the exe is now named Portable.exe. I'm just searching for a Way to use `WindowsBuild/Portable' as output directory but keeping the rest as it is.

**Comment by rokm** [comment]
The `--name` argument affects both the `name` argument to the `EXE` and to the `COLLECT` entry in the generated spec file.

So if I understand correctly what you are after, you would need to build using the .spec file instead from .py, and change the `name` of `COLLECT` to `Portable`, while leaving the `name` of `EXE` at `<scriptname>`. Then, build from spec with `--distpath WindowsBuild`.

But then again, why not just rename the `dist/<scriptname>` that PyInstaller generates by default to what you want them to be (i.e., as a step on your CI that follows the PyInstaller build step)?

**Reply by JakobDev** [reply]
I just want a Way to save the files in `WindowsBuild/Portable` instead of `dist/<scriptname>`

> But then again, why not just rename

That's what I#m currently doing, but I just wonder is there is a better Way
